Replace/upgrade virtual scrollers

bhollis opened the pull request:
1. Replace `react-virtualized` with `@tanstack/react-virtual`. It's a lot smaller and easier to use.
2. Item heights are now computed dynamically, on demand. No more "render an example row". I managed to get this working using `react-virtualized` (first commit) but `@tanstack/react-virtual` supports it better (it uses resize observers to keep track of sizes if they change). `@tanstack/react-virtual` also uses a technique where it moves along a container item with the windowed rows using `translateY` instead of individually sizing and positioning each item.
3. Apply this, via some common utility components, to LO sets, Loadouts, Item Feed, and even the Inventory screen (too much?). No more managing `itemsToShow` in Item Feed. This work replaces #9512.

On the item feed at least, there is some minor repositioning that's noticeable if you scroll fast enough as the library catches up with measuring each item, but I think it's still nicer than the old method of rendering an ever-increasing list as you scroll. I'd love some testing on other computers/browsers as I rarely experience performance issues.

bhollis commented on src/app/item-feed/ItemFeed.tsx:
```diff
@@ -35,13 +29,7 @@ const Item = memo(function Item({ item, tag }: { item: DimItem; tag: TagValue |
     </ItemPopupTrigger>
   );
   return (
-    <motion.div
-      className={styles.item}
-      initial={{ scale: 0, opacity: 0 }}
-      exit={{ scale: 0, opacity: 0 }}
-      animate={{ scale: 1, opacity: 1 }}
-      transition={spring}
-    >
```
I gave up on having items animate in and out in the item feed. It was fun, but it becomes way more obvious when you're scrolling and items are appearing/disappearing constantly.

robojumper reviewed (changes requested):
1. On Windows 10 Edge and Firefox this breaks drag and drop on the main inventory page.
2. As noted in #9512, this breaks the "click on menu entry to scroll to loadout" behavior on the desktop Loadouts page -- this might need to be wired up with virtual's [scrollToIndex](https://tanstack.com/virtual/v3/docs/api/virtualizer#scrolltoindex)

robojumper commented:
Very happy to see the set height measuring code gone. Item Feed feels much snappier now when scrolling quickly.

bhollis commented:
OK, I've fixed the Loadouts menu, and I just reverted using this for the Inventory page. My guess is the fact that the elements are now inside a transformed div messes up the drag and drop positioning calculations. Weirdly, drag and drop does still work on the Loadouts screen...
